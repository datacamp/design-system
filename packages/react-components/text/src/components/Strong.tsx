import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

import baseStyle from '../baseStyle';
import computeDataAttributes from '../computeDataAttributes';

interface StrongProps {
  children: string;
  className?: string;
  dataAttributes?: { [key: string]: string };
}

const strongStyle = css(baseStyle, {
  fontWeight: tokens.size.fontWeight.bold.value,
  lineHeight: tokens.size.font.base.value,
});

const Strong: React.FC<StrongProps> = ({
  children,
  className,
  dataAttributes,
}) => {
  const parsedDataAttributes = computeDataAttributes(dataAttributes);

  return (
    <strong className={className} css={strongStyle} {...parsedDataAttributes}>
      {children}
    </strong>
  );
};

export default Strong;
