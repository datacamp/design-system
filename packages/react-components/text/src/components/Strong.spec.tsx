import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import React from 'react';

import Strong from './Strong';

const testText = 'Some example text.';
const testClassName = 'example-class';
const testDataAttributes = { example: 'testdata', item: 'othertest' };

describe('<Strong />', () => {
  it('renders', async () => {
    const { container } = await axeRender(
      <Strong className={testClassName} dataAttributes={testDataAttributes}>
        {testText}
      </Strong>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('sets the text in the span', async () => {
    const { getByText } = await axeRender(<Strong>{testText}</Strong>);
    expect(getByText(testText)).toBeDefined();
  });

  it('sets the className', async () => {
    const { getByText } = await axeRender(
      <Strong className={testClassName}>{testText}</Strong>,
    );
    expect(getByText(testText)).toHaveClass(testClassName);
  });

  it('sets the data attributes', async () => {
    const { getByText } = await axeRender(
      <Strong dataAttributes={testDataAttributes}>{testText}</Strong>,
    );
    expect(getByText(testText)).toHaveAttribute(
      'data-item',
      testDataAttributes.item,
    );
    expect(getByText(testText)).toHaveAttribute(
      'data-example',
      testDataAttributes.example,
    );
  });
});
