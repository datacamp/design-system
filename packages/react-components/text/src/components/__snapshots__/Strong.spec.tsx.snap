// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Strong /> renders 1`] = `
.emotion-0 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: sans-serif;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  font-weight: 800;
  line-height: 16px;
}

<strong
  class="example-class emotion-0"
  data-example="testdata"
  data-item="othertest"
>
  Some example text.
</strong>
`;
